File: internal/templates/data/common/py/boot.py
import logging
import os
import sys
from agentuity import autostart

if __name__ == "__main__":
    logging.basicConfig(
        level=logging.INFO,
        format='[%(levelname)-5.5s] %(message)s',
    )
    
    # Check if AGENTUITY_API_KEY is set
    if not os.environ.get('AGENTUITY_API_KEY'):
        print("\033[31m[ERROR] AGENTUITY_API_KEY is not set. This should have been set automatically by the Agentuity CLI or picked up from the .env file.\033[0m")
        sys.exit(1)
    
    # Check if AGENTUITY_URL is set
    if not os.environ.get('AGENTUITY_URL'):
        print("\033[31m[WARN] You are running this agent outside of the Agentuity environment. Any automatic Agentuity features will be disabled.\033[0m")
        print("\033[31m[WARN] Recommend running `agentuity dev` to run your project locally instead of python script.\033[0m")
    
    autostart()

File: internal/templates/data/common/js/boot.ts
import { runner } from "@agentuity/sdk";

if (!process.env.AGENTUITY_API_KEY) {
	console.error("AGENTUITY_API_KEY is not set. This should have been set automatically by the Agentuity CLI or picked up from the .env file.");
	process.exit(1);
}

runner(true, import.meta.dirname).catch((err) => {
	if (err instanceof Error) {
		console.error(err.message);
		console.error(err.stack);
	} else {
		console.error(err);
	}
	process.exit(1);
});